exaudfclient/base/script_options_parser/ctpg/test/script_option_lines_test.cpp

```diff
@@ -223,3 +223,75 @@ TEST(ScriptOptionLinesTest, test_multiple_lines_with_code) {
     ASSERT_EQ(jar_option_result->second.size(), 1);
     ASSERT_EQ(jar_option_result->second[0], buildOption("/buckets/bucketfs1/jars/exajdbc.jar", 49, 41));
 }
+
+
+class ScriptOptionLinesEscapeSequenceTest : public ::testing::TestWithParam<std::pair<std::string, std::string>> {};
+
+TEST_P(ScriptOptionLinesEscapeSequenceTest, test_escape_seq_in_option_value) {
+    const std::pair<std::string, std::string> escape_seq = GetParam();
+    /**
+    Verify that the parser replaces escape sequences correctly.
+    */
+    const std::string code =
+        "%jvmoption -Dhttp.agent=ABC" + escape_seq.first + "DEF; class Abc{};\n"
+        "%jar /buckets/bucketfs1/jars/exajdbc.jar; class DEF{};\n";
+
+    options_map_t result;
+    parseOptions(code, result, throwException);
+    ASSERT_EQ(result.size(), 2);
+
+    const auto jvm_option_result = result.find("jvmoption");
+    ASSERT_NE(jvm_option_result, result.end());
+    ASSERT_EQ(jvm_option_result->second.size(), 1);
+    EXPECT_EQ(jvm_option_result->second[0].value, std::string("-Dhttp.agent=ABC" + escape_seq.second + "DEF"));
+
+    const auto jar_option_result = result.find("jar");
+    ASSERT_NE(jar_option_result, result.end());
+    ASSERT_EQ(jar_option_result->second.size(), 1);
+    ASSERT_EQ(jar_option_result->second[0].value, "/buckets/bucketfs1/jars/exajdbc.jar");
+}
+
+/*
+ '\n' -> new line character
+ '\r' -> return character
+ '\;' -> semicolon
+ '\a' -> anything else should not be replaced.
+ */
+const std::vector<std::pair<std::string, std::string>> escape_sequences =
+        {std::make_pair("\\n", "\n"), std::make_pair("\\r", "\r"), std::make_pair("\\;", ";"), std::make_pair("\\a", "\\a")};
+
+INSTANTIATE_TEST_SUITE_P(
+    ScriptOptionLines,
+    ScriptOptionLinesEscapeSequenceTest,
+    ::testing::ValuesIn(escape_sequences)
+);
+
+class ScriptOptionLinesRestTest : public ::testing::TestWithParam<std::string> {};
+
+TEST_P(ScriptOptionLinesRestTest, test_rest_with_tokens) {
+    const std::string rest = GetParam();
+    /**
+    Verify that the parser correctly ignores character sequences containing special parser tokens
+    after the options in a line.
+    */
+    const std::string code =
+        "%jvmoption -Dhttp.agent=abc; class Abc{};" + rest;
+
+    options_map_t result;
+    parseOptions(code, result, throwException);
+    ASSERT_EQ(result.size(), 1);
+
+    const auto jvm_option_result = result.find("jvmoption");
+    ASSERT_NE(jvm_option_result, result.end());
+    ASSERT_EQ(jvm_option_result->second.size(), 1);
+    ASSERT_EQ(jvm_option_result->second[0], buildOption("-Dhttp.agent=abc", 0, 28));
```

Comment:
These asserts seem only to test the presence of the defined options, but not the absence of not intended options.